src/invocation/InvocationService.ts

```diff
@@ -24,13 +25,16 @@ class InvocationService {
         return deferred.promise;
     }
 
-    public invokeOnPartition(clientMessage: ClientMessage, partitionId: number): Q.Promise<ClientMessage> {
+    invokeOnPartition(clientMessage: ClientMessage, partitionId: number): Q.Promise<ClientMessage> {
         clientMessage.setPartitionId(partitionId);
-        var connection: ClientConnection = this.client.getConnectionManager().getOwnerConnection();
-        return this.invokeOnConnection(connection, clientMessage);
+        var address: Address = this.client.getPartitionService().getAddressForPartition(partitionId);
+        return this.client.getConnectionManager().getOrConnect(address)
+            .then<ClientMessage>((connection: ClientConnection) => {
+                return this.invokeOnConnection(connection, clientMessage);
+            });
     }
 
-    public processResponse(buffer: Buffer) {
+    processResponse(buffer: Buffer) {
         var clientMessage = new ClientMessage(buffer);
         var correlationId = clientMessage.getCorrelationId().toNumber();
         this.pending[correlationId].resolve(clientMessage);
```

Comment:
We should check if `clientMessage` is an event by checking event flag. If it is not an event, we should remove that entry with `correlationId` from `this.pending`.